runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -6413,7 +6413,20 @@ static TR::Register* inlineIntrinsicIndexOf(TR::Node* node, TR::CodeGenerator* c
    TR::Register *resultReg = endReg;
    TR_Debug *debugObj = cg->getDebug();
 
-   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, dataAddrReg, arrayReg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+
+#ifdef J9VM_GC_SPARSE_HEAP_ALLOCATION
+   if (TR::Compiler->om.isOffHeapAllocationEnabled())
+      {
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, dataAddrReg, TR::MemoryReference::createWithDisplacement(cg, arrayReg, cg->comp()->fej9()->getOffsetOfContiguousDataAddrField()));
+      generateCompareImmInstruction(cg, node, dataAddrReg, 0, /* is64bit */ true);
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, doneLabel, TR::CC_EQ);
```

Comment:
I think you need to initialize `resultReg` to something before this branch is taken. Branching to `doneLabel` will immediately result in `resultReg` being tested and set via the `csinvx` instruction.